### DISCUSSION TITLE
Inquietud en calificación grade 0

### DISCUSSION THREAD

**Discussion by leomare12** [rramosp/ai4eng.v1#233] [02 - PYTHON]
Hola, este error ...

"task_03 submitted. your grade is 0
----- grader message -------
testing with 100 random calls
error cannot do the operation over the same input object.
you must make a copy of the input object X = X.copy()
or perform operations that implicitly make the copy.
observe that
X += 1
adds one to a matrix in place (modifying the matrix X)
whereas
X = X+1
creates a new matrix and assigns the name X to it."

Es algo no permitido o es el calificador que no permite que se reemplace la matriz de trabajo.
lo hice porque en este caso la matriz X base no la necesitaba después.

ya fue corregido creando una nueva matriz, pero me queda esa duda.

**Comment by Felipe-RA** [comment]
En el ejercicio buscamos que no modifiquen la matriz original.

Esto es recomendable por varios motivos, entre ellos:

- Evitas "dañar los datos" y tienes a mano un respaldo.
- Si modificas la matriz original debes volver a cargar los datos cada vez que quieras probar un cambio distinto.
